src/linter.ts

```diff
@@ -142,6 +142,10 @@ export class Linter {
     }
 
     public getResult(): LintResult {
+        if (this.options.quiet) {
+          this.failures = this.failures.filter((failure) => failure.getRuleSeverity() === "error");
```

Comment:
Can you use a new variable here rather than overwriting this.failures - perhaps something like:

```
const errors = this.failures.filter((failure) => failure.getRuleSeverity() === "error");
const failures = this.options.quiet ? errors : this.failures;
```

Reply:
Sure, done!